### PR TITLE
parser: handle the national character set like N'literal'

### PR DESCRIPTION
**Pull request by tiancaiamao**
```
mysql> select N'abc';
ERROR 1105 (HY000): unknown column n
```

mysql N'xxx' or n'xxx' is the same as _utf8'xxx'

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @shenli

File: parser/lexer.go [hunk 1]
```diff
@@ -222,6 +222,20 @@ func startWithXx(s *Scanner) (tok int, pos Pos, lit string) {
 	return
 }
 
+func startWithNn(s *Scanner) (tok int, pos Pos, lit string) {
+	tok, pos, lit = scanIdentifier(s)
+	// The National Character Set, N'some text' or n'some test'.
+	// See https://dev.mysql.com/doc/refman/5.7/en/string-literals.html
+	// and https://dev.mysql.com/doc/refman/5.7/en/charset-national.html
+	if lit == "N" || lit == "n" {
+		if s.r.peek() == '\'' {
```
Comment (shenli) [inline parser/lexer.go]: Is N"xxx" also underscoreCS?

Comment (tiancaiamao) [inline parser/lexer.go]: N"xxx" is parsed as two tokens: an underscoreCS and a stringLit, and they are handled in parser.y later.

Comment (tiancaiamao) [inline parser/lexer.go]: Normally, `N` is parsed to identifier, but when `N` is followed by `"` without whitespace, try to parse it to underscoreCS

**Comment by shenli** [comment]
LGTM
@coocood PTAL

File: parser/lexer.go (hunk 1 above)
Comment (tiancaiamao) [inline parser/lexer.go]: I make a mistake.
N"xxx" is not underscoreCS
N'xxx' is underscoreCS

File: parser/misc.go [hunk 2]
```diff
@@ -125,10 +125,11 @@ func init() {
 	initTokenFunc("-", startWithDash)
 	initTokenFunc("#", startWithSharp)
 	initTokenFunc("Xx", startWithXx)
+	initTokenFunc("Nn", startWithNn)
 	initTokenFunc("x", startWithXx)
 	initTokenFunc("b", startWithb)
```
Comment (coocood) [inline parser/misc.go]: This should be "Bb"?

File: parser/misc.go [hunk 3]
```diff
@@ -125,10 +125,11 @@ func init() {
 	initTokenFunc("-", startWithDash)
 	initTokenFunc("#", startWithSharp)
 	initTokenFunc("Xx", startWithXx)
+	initTokenFunc("Nn", startWithNn)
 	initTokenFunc("x", startWithXx)
```
Comment (coocood) [inline parser/misc.go]: This can be removed?

Comment (tiancaiamao) [inline parser/misc.go]: startWithXx may return hexLit or identifier
startWithb may return bitLit or identidier

**Pull request merged**